### PR TITLE
NP-46371 Add checkbox for nvi reported filter

### PR DESCRIPTION
**Pull request by iBuzza**
# Description

Added new parameter to Advanced Search: scientificIndexStatus.

The paramater is currently not giving any hits in dev, but works in test:

https://api.dev.nva.aws.unit.no/search/resources?aggregation=none&scientificIndexStatus=Reported&from=0&results=10&order=modifiedDate&sort=desc

https://api.test.nva.aws.unit.no/search/resources?aggregation=none&scientificIndexStatus=Reported&from=0&results=10&order=modifiedDate&sort=desc

Link to Jira issue:

https://unit.atlassian.net/jira/software/c/projects/NP/boards/111?assignee=6361029ec97f5473af719859&selectedIssue=NP-46371

# Checklist

## Required

- [x] The changes are working as expected (see description)
- [x] The changes are tested OK for both desktop and mobile screens
- [x] The changes are tested OK for a11y (with [axe](https://www.deque.com/axe/devtools/), [lighthouse](https://developer.chrome.com/docs/lighthouse/), or similar)
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

## Optional

- [x] Solution is verified by PO/designer

### CODE REVIEW

File: src/api/searchApi.ts [hunk 1]
```diff
@@ -216,6 +217,7 @@ export interface FetchResultsParams {
   [ResultParam.Results]?: number | null;
   [ResultParam.Series]?: string | null;
   [ResultParam.ScientificIndex]?: string | null;
+  [ResultParam.ScientificIndexStatus]?: string | null;
```
Comment (elkebab) [inline src/api/searchApi.ts:220]: burde denne også ha type? antar den støtter et fixed sett med argument
```suggestion
  [ResultParam.ScientificIndexStatus]?: ScientificIndexStatuses | null;
```

File: src/pages/search/advanced_search/AdvancedSearchPage.tsx [hunk 2]
```diff
@@ -152,6 +176,19 @@ export const AdvancedSearchPage = () => {
           <StyledTypography fontWeight="bold">{t('common.language')}</StyledTypography>
           <LanguageFilter />
         </Grid>
+
+        {showFilterDivider && <StyledDivider orientation="vertical" flexItem />}
+
+        <Grid item>
```
Comment (elkebab) [inline src/pages/search/advanced_search/AdvancedSearchPage.tsx:182]: denne har to items i seg (`<StyledTypography>` og `<FormControlLabel>`), er det ikke da en container? 🤔 
```suggestion
        <Grid container item>
```

Comment (iBuzza) [reply src/pages/search/advanced_search/AdvancedSearchPage.tsx:182]: Vi har ikke gjort det på de andre feltene 🤔 Må jo da begynne å definere `direction` osv, så vet ikke om det er verdt innsatsen? Med mindre du tenker at alle `children` telles som `item` uansett?

Comment (elkebab) [reply src/pages/search/advanced_search/AdvancedSearchPage.tsx:182]: Som diskutert: missforstelse fra meg!

File: src/pages/search/advanced_search/AdvancedSearchPage.tsx [hunk 3]
```diff
@@ -64,6 +77,7 @@ export const AdvancedSearchPage = () => {
     publicationYearSince: params.get(ResultParam.PublicationYearSince),
     publisher: params.get(ResultParam.Publisher),
     results: Number(params.get(ResultParam.Results) ?? ROWS_PER_PAGE_OPTIONS[0]),
+    scientificIndexStatus: params.get(ResultParam.ScientificIndexStatus) as ScientificIndexStatuses,
```
Comment (elkebab) [inline src/pages/search/advanced_search/AdvancedSearchPage.tsx:80]:
```suggestion
    scientificIndexStatus: params.get(ResultParam.ScientificIndexStatus) as ScientificIndexStatuses | null,
```

**Review by elkebab — APPROVED** [review]
Har ikke testet, men koden ser fin ut 👍

**Pull request merged**